### DISCUSSION TITLE
Model Breaks down on using self loops

### DISCUSSION THREAD

**Discussion by SourabhRanade** [pyg-team/pytorch_geometric#7222] [Q&A] [answered]
I have a model (consisting of a few `GraphSage` Layers followed by some `torch.nn.Linear` layers)  that works perfectly fine when I **train** it on differently sized fully connected graphs without **self loops**. However, as soon as I add self loops to the graph using  `add_self_loops` method and start the training, the model basically stops learning. I am not very sure why. 
I also tried creating the self loops while I build my graphs (using `networkx` thus not using the `add_self_loops` method). I still end up with the model not learning anything. Each epoch basically gives me similar losses and accuracies. 

Can someone guide me with this?  Thank you in advance.

**Comment by rusty1s** [comment, marked as answer]
This is hard to tell. In the end, adding a self-loop as part of the graph means that your GNN can no longer differentiate neighbors from the central node, which may be a problem for your specific task/data.
